libvast_test/CMakeLists.txt

```diff
@@ -0,0 +1,97 @@
+## Configure data.hpp
+
+# Bro logs.
+set(test_data "#ifndef TEST_DATA\n#define TEST_DATA")
+set(test_data "${test_data}\n\nnamespace bro {\n")
+file(GLOB logs RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" artefacts/logs/bro/*)
+foreach (log ${logs})
+  get_filename_component(log_basename ${log} NAME_WE)
+  set (this "constexpr auto ${log_basename} = VAST_TEST_PATH\"${log}\";")
+  set (test_data "${test_data}\n${this}")
+endforeach ()
+set(test_data "${test_data}\n\n} // namespace bro")
+
+# Bgpdump logs.
+set(test_data "${test_data}\n\nnamespace bgpdump {\n")
+file(GLOB logs RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" artefacts/logs/bgpdump/*.txt)
+foreach (log ${logs})
+  get_filename_component(log_basename ${log} NAME_WE)
+  set (this "constexpr auto ${log_basename} = VAST_TEST_PATH\"${log}\";")
+  set (test_data "${test_data}\n${this}")
+endforeach ()
+set(test_data "${test_data}\n\n} // namespace bgpdump")
+
+# MRT logs.
+set(test_data "${test_data}\n\nnamespace mrt {\n")
+file(GLOB logs RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" artefacts/logs/mrt/*)
+foreach (log ${logs})
+  get_filename_component(log_basename ${log} NAME_WE)
+  set (this "constexpr auto ${log_basename} = VAST_TEST_PATH\"${log}\";")
+  set (test_data "${test_data}\n${this}")
+endforeach ()
+set(test_data "${test_data}\n\n} // namespace mrt")
+
+# PCAP traces.
+set(test_data "${test_data}\n\nnamespace traces {\n")
+file(GLOB traces RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" artefacts/traces/*.pcap)
+foreach (trace ${traces})
+  get_filename_component(trace_basename ${trace} NAME_WE)
+  set(this "constexpr auto ${trace_basename} = VAST_TEST_PATH\"${trace}\";")
+  set(test_data "${test_data}\n${this}")
+endforeach ()
+set(test_data "${test_data}\n\n} // namespace traces")
+set(test_data "${test_data}\n\n#endif")
+
+configure_file(vast/test/data.hpp.in ${CMAKE_CURRENT_BINARY_DIR}/vast/test/data.hpp)
```

Comment:
I suppose it does the trick for now, but we I'm really looking forward to get rid of this and instead generate `.hpp` files from the test logs directly.

Reply:
dito